packages/rum-core/src/performance-monitoring/capture-hard-navigation.js

```diff
@@ -51,35 +54,27 @@ const eventPairs = [
   ['loadEventStart', 'loadEventEnd', 'Fire "load" event']
 ]
 
-function isValidSpan(transaction, span) {
-  const duration = span.duration()
-  return (
-    duration < SPAN_THRESHOLD &&
-    duration > 0 &&
-    span._start <= transaction._end &&
-    span._end <= transaction._end
-  )
-}
-
-function isValidPerformanceTiming(start, end, baseTime = 0) {
+function shouldCreateSpan(start, end, baseTime, transactionEnd) {
   return (
     typeof start === 'number' &&
     typeof end === 'number' &&
     start >= baseTime &&
     end > start &&
-    end - start < SPAN_THRESHOLD &&
-    start - baseTime < SPAN_THRESHOLD &&
-    end - baseTime < SPAN_THRESHOLD
+    end - baseTime < transactionEnd &&
+    end - start < MAX_SPAN_DURATION &&
+    start - baseTime < MAX_SPAN_DURATION &&
+    end - baseTime < MAX_SPAN_DURATION
   )
 }
 
-function createNavigationTimingSpans(timings, baseTime) {
+function createNavigationTimingSpans(timings, transactionEnd) {
   const spans = []
+  const baseTime = timings.fetchStart
```

Comment:
Yeah because its available already as part of timings object and having it as a added parameter was not required. We already know that its gonna be fetchStart/navigationStart if we decide to change in future. But just unnecessary param IMO.

Reply:
The baseTime doesn't necessarily have to be part of timings, we can choose a different baseTime (some providers do this in fact), furthermore this reduces the testability of this function since we can not provide a different baseTime to make sure it performs correctly in all cases.